core/src/main/java/org/elasticsearch/search/slice/SliceBuilder.java

```diff
@@ -194,9 +196,14 @@ public Query toFilter(QueryShardContext context, int shardId, int numShards) {
             throw new IllegalArgumentException("field " + field + " not found");
         }
 
+        String field = this.field;
         boolean useTermQuery = false;
         if (UidFieldMapper.NAME.equals(field)) {
-           useTermQuery = true;
+            if (context.getIndexSettings().isSingleType()) {
```

Comment:
I wonder whether there is an expectation that a given seed always gives the same score to a given id, regardless of whether it was updated. I remember Robert arguing that we should just use the Lucene doc id for random score generation.